drivers/display/display_st7789v.c

```diff
@@ -67,29 +68,42 @@ static void st7789v_set_lcd_margins(const struct device *dev,
 	data->y_offset = y_offset;
 }
 
-static void st7789v_set_cmd(const struct device *dev, int is_cmd)
-{
-	const struct st7789v_config *config = dev->config;
-
-	gpio_pin_set_dt(&config->cmd_data_gpio, is_cmd);
-}
-
 static void st7789v_transmit(const struct device *dev, uint8_t cmd,
 			     uint8_t *tx_data, size_t tx_count)
 {
 	const struct st7789v_config *config = dev->config;
+	size_t index;
+	uint16_t data = cmd;
 
 	struct spi_buf tx_buf = { .buf = &cmd, .len = 1 };
 	struct spi_buf_set tx_bufs = { .buffers = &tx_buf, .count = 1 };
 
-	st7789v_set_cmd(dev, 1);
-	spi_write_dt(&config->bus, &tx_bufs);
+	if (config->cmd_data_gpio.port != NULL) {
+		if (cmd != ST7789V_CMD_NONE) {
+			gpio_pin_set_dt(&config->cmd_data_gpio, 1);
+			spi_write_dt(&config->bus, &tx_bufs);
+		}
+
+		if (tx_data != NULL) {
+			tx_buf.buf = tx_data;
+			tx_buf.len = tx_count;
+			gpio_pin_set_dt(&config->cmd_data_gpio, 0);
+			spi_write_dt(&config->bus, &tx_bufs);
+		}
+	} else {
+		tx_buf.buf = &data;
+		tx_buf.len = 2;
+
+		if (cmd != ST7789V_CMD_NONE) {
+			spi_write_dt(&config->bus, &tx_bufs);
+		}
 
-	if (tx_data != NULL) {
-		tx_buf.buf = tx_data;
-		tx_buf.len = tx_count;
-		st7789v_set_cmd(dev, 0);
-		spi_write_dt(&config->bus, &tx_bufs);
+		if (tx_data != NULL) {
+			for (index = 0; index < tx_count; ++index) {
```

Comment:
`		for (size_t index = 0; index < tx_count; ++index) {` ? since index is only used in the loop